src/API/QueryFieldLocationService.php

```diff
@@ -0,0 +1,18 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+namespace EzSystems\EzPlatformQueryFieldType\API;
+
+use eZ\Publish\API\Repository\Values\Content\Location;
+
+interface QueryFieldLocationService
+{
+    public function loadContentItemsForLocation(Location $location, string $fieldDefinitionIdentifier): iterable;
+
+    public function loadContentItemsSliceForLocation(Location $location, string $fieldDefinitionIdentifier, int $offset, int $limit): iterable;
+
+    public function countContentItemsForLocation(Location $location, string $fieldDefinitionIdentifier): int;
+}
```

Comment:
In general I'd love some description of what this does, the way you've described it to me during internal sync (so how `$location` is used and what (briefly) is performed and returned). Since all the methods follow the same pattern, common description on the interface would be fine. The drawback in this case is that developers are more likely to look for the doc of method, not the injected service.

Though some more strict type hint for `load*` methods would be nice, as `iterable` is too vague.

Reply:
Of course, I will make sure there is one.